pkg/dmaasbackup/schedule.go

```diff
@@ -72,12 +77,84 @@ func (d *dmaasBackup) generateBackupName(dbkp v1alpha1.DMaaSBackup) string {
 // updateScheduleInfo checks for relevant velero schedule/backup and
 // update the dbkp with schedule/backup details
 func (d *dmaasBackup) updateScheduleInfo(dbkp *v1alpha1.DMaaSBackup) error {
-	// TODO
+	var (
+		schedule *velerov1api.Schedule
+		bkp      *velerov1api.Backup
+		err      error
+	)
+
+	// check if dbkp is having config for non-scheduled backup
+	if yes := isSchedule(*dbkp); !yes {
+		bkp, err = d.backupClient.Get(dbkp.Name, metav1.GetOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			return errors.Wrapf(err, "failed to update scheduleInfo")
+		}
+		if bkp.Name == "" {
+			// Backup could have deleted due to expiration of TTL
+			// TODO: Should we clear backupname from dmaasbackup? and/or recreate?
+			return nil
+		}
+		if dbkp.Status.VeleroBackupName == nil {
+			backupName := bkp.Name
+			dbkp.Status.VeleroBackupName = &backupName
+		}
+		return nil
+	}
+
+	// check for any queued dummy entry in veleroschedule
+	dummySchedule := getDummyVeleroSchedule(dbkp)
+
+	if dummySchedule == nil {
+		// we haven't planned any schedule creation
+		goto lastschedule_cleanup
+	}
+
+	// dummy entry exists in veleroschedule
+	// check for any stale schedules
+
+	// for scheduled backup
+	schedule, err = d.scheduleClient.Get(dummySchedule.ScheduleName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// no schedule exists with dummy entry's name
+			return nil
+		}
+		return errors.Wrapf(err, "failed to update scheduleInfo")
+	}
+
+	// we found stale schedule
+	// update dummy entry with schedule details
+	updateDummyVeleroSchedule(dbkp, dummySchedule, schedule)
+
+lastschedule_cleanup:
+	// check if we have any last schedule
+	lastSchedule := getLastVeleroSchedule(dbkp)
```

Comment:
why it is deleting last schedule?

Reply:
ok.. it is returning the previous Velero Schedule due to reverse order of storing the Schedules